editor/src/components/user-bar.tsx

```diff
@@ -53,7 +56,21 @@ export const UserBar = React.memo(() => {
 })
 UserBar.displayName = 'UserBar'
 
-export const SinglePlayerUserBar = React.memo(() => {
+const SinglePlayerUserBar = React.memo(() => {
+  const dispatch = useDispatch()
+
+  const url = window.location.href
+  const handleCopyToClipboard = React.useCallback(async () => {
+    try {
+      let actions: EditorAction[] = []
+      actions.push(showToast(notice('Project link copied to clipboard!', 'NOTICE', false)))
+      await window.navigator.clipboard.writeText(url)
+      dispatch(actions)
```

Comment:
Nitpick: you don't need to create an array here, we only do that in edge case situations where we conditionally push actions into the array, but want to keep it a single dispatch.

For this simple case, you can directly dispatch the action:

```suggestion
      await window.navigator.clipboard.writeText(url)
      dispatch([showToast(notice('Project link copied to clipboard!', 'NOTICE', false))])
```